app/src/main/java/com/nitramite/utils/BackupUtils.java

```diff
@@ -0,0 +1,182 @@
+        }
+    }
+
+
+    /**
+     * Get database file input stream object
+     *
+     * @param context Context
+     * @return file input stream
+     * @throws FileNotFoundException database file not found
+     */
+    private static FileInputStream getDBFileInputStream(Context context, Backup backup) throws FileNotFoundException {
+        @SuppressLint("SdCardPath") final String databaseLocation = "/data/data/" + context.getApplicationContext().getPackageName() + "/databases/" + DATABASE_NAME;
+        File databaseFile = new File(databaseLocation);
+        backup.setFileName(DATABASE_NAME);
+        return new FileInputStream(databaseFile);
+    }
+
+
+    /**
+     * Get database file output stream object
+     *
+     * @param context Context
+     * @return file output stream
+     * @throws FileNotFoundException todo: document uri not found?
+     */
+    private static OutputStream getDBOutputStream(Context context, Backup backup) throws FileNotFoundException, NullPointerException {
+        OutputStream outputStream = null;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            ContentResolver resolver = context.getContentResolver();
+            ContentValues contentValues = new ContentValues();
+            contentValues.put(MediaStore.MediaColumns.DISPLAY_NAME, DATABASE_NAME);
+            contentValues.put(MediaStore.MediaColumns.MIME_TYPE, "application/x-sqlite3");
+            contentValues.put(MediaStore.MediaColumns.RELATIVE_PATH, Environment.DIRECTORY_DOWNLOADS);
+            Uri documentUri = resolver.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues);
+            outputStream = resolver.openOutputStream(Objects.requireNonNull(documentUri));
+            backup.setLocation(context.getString(R.string.backup_location_external_storage_downloads_dir));
+        } else {
+            @SuppressWarnings("deprecation") String downloadsDir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).toString();
+            File file = new File(downloadsDir, DATABASE_NAME);
+            outputStream = new FileOutputStream(file);
+            backup.setLocation(context.getString(R.string.backup_location_external_storage_downloads_dir));
+        }
+        return outputStream;
+    }
+
+
+    /**
+     * Get file input stream from db source path
+     *
+     * @param context Context
+     * @return file input steam
+     * @throws FileNotFoundException db file not found
+     */
+    private static FileInputStream getDBFileInputSteamFromExternalStorage(Context context) throws FileNotFoundException, NullPointerException {
+        FileInputStream fileInputStream = null;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+
+            // TODO: Wrong implementation because Google's documentation sucks with this subject.
+            // TODO: currently using android:requestLegacyExternalStorage="true" at manifest.
+            // TODO: research proper solution!
```

Comment:
Adding a separate `TODO` for each line will generate more todo's in the IDE's list: :stuck_out_tongue:

Reply:
Have not been doing multiline todo before... fixed to be one TODO still three lines.